File: ELearningFrontEnd/src/app/model/lecture.ts
import {Chapter} from "./chapter";

export interface Lecture{

    id: number | null;
    name: string | null;
    videoUrl: string | null;
    resourceUrl: string | null;
    preview: boolean | null;
    chapter: Chapter | null;
    activated: boolean | null,
    createdTime: Date | null,
    updatedTime: Date | null,
    approvalStatus: string | null,
    orderNumber: number | null
    videoDuration: number | null
}

File: ELearningBackEnd/src/main/java/com/example/elearningbackend/lecture/Lecture.java
package com.example.elearningbackend.lecture;

import com.example.elearningbackend.chapter.Chapter;
import com.example.elearningbackend.common.FullEntity;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "lectures")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Lecture extends FullEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(nullable = false)
    private String name;


    private String videoUrl;


    private String resourceUrl;

    @ManyToOne
    @JoinColumn(name = "chapter_id")
    private Chapter chapter;

    @Column
    private boolean isPreview;

    private String approvalStatus;

    private int orderNumber;

    private Float videoDuration;

    private long countLearnerComplete ;
}
